transmogrifier/sources/xml/dspace_mets.py

```diff
@@ -212,6 +136,136 @@ def get_content_type(cls, source_record: Tag) -> list[str] | None:
             for content_type in source_record.find_all("mods:genre", string=True)
         ] or None
 
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        contributors = []
+        for contributor in source_record.find_all("mods:name"):
+            if name := contributor.find("mods:namePart", string=True):
+                if role := contributor.find("mods:roleTerm", string=True):
+                    kind = str(role.string)
+                else:
+                    kind = "Not specified"
+                contributors.append(
+                    timdex.Contributor(
+                        kind=kind,
+                        value=str(name.string),
+                    )
+                )
+        return contributors or None
+
+    @classmethod
+    def get_dates(cls, source_record: Tag) -> list[timdex.Date] | None:
+        # Only publication date is mapped from DSpace, other relevant date field (dc.
+        # coverage.temporal) is not mapped to the OAI-PMH METS output.
+        if publication_date := source_record.find("mods:dateIssued", string=True):
+            publication_date_value = str(publication_date.string.strip())
+            if validate_date(
+                publication_date_value, cls.get_source_record_id(source_record)
+            ):
+                return [
+                    timdex.Date(kind="Publication date", value=publication_date_value)
+                ]
+        return None
+
+    @classmethod
+    def get_file_formats(cls, source_record: Tag) -> list[str] | None:
+        # Only maps formats with attribute use="ORIGINAL" because other formats such as
+        # USE="TEXT" are used internally by DSpace and not made publicly available.
+        file_formats = []
+        for file_group in source_record.find_all("fileGrp", USE="ORIGINAL"):
```

Comment:
Does `USE` need to be uppercase?  Not a big deal, just surprising to see it that way.

Reply:
Nevermind!  Keeping comment for historical reviewing purposes, but I see now that the attribute `USE` is uppercase in the DSpace METS.